### PR TITLE
Improve crayon UI to not be stuck in 1996

### PR DESCRIPTION
**Pull request by SaphireLattice**
## About the PR
Crayon

Makes the crayon UI not feel like a placeholder. Adds categories to it all that you don't have to filter out letters from various lines. Changes the search input to automatically select whatever decal with ID matching the search exactly. And then makes the search double up as a queue where the individual searches are separated by comma. And it will automatically advance when you draw with the crayon (if you draw a symbol that doesn't match searched ID, it will not advance though).

## Why / Balance
Crayon

The current UI is bad, and the only reason it's remotely usable is the search box. Everything is mixed around without regard to any sort of categorisation.

## Media
<details>
<summary>Crayon</summary>

image

</details>

## Technical details
Added Linq to the Xaml, ignored a deprecation warning for the sprite getting thing, abused sorting. Crayon.

## Requirements

- [X] I have read and am following the [Pull Request and Changelog Guidelines](https://docs.spacestation14.com/en/general-development/codebase-info/pull-request-guidelines.html).
- [X] I have added media to this PR or it does not require an ingame showcase.

**Changelog**

:cl:
- add: Crayon UI now has categories and queue

### CODE REVIEW

**Comment by SaphireLattice** [comment]
Crayon


https://github.com/user-attachments/assets/e0ed3e19-eaff-4682-8ab2-dc049b0ec501

**Comment by SlamBamActionman** [comment]
Okay it's very funny to say crayon but I really would prefer if you followed PR guidelines and explained the features your PR is implementing. Based on the videos it seems like it's some cool stuff!

Does the new UI play well with the rainbow crayon?

**Comment by Everturning** [comment]
Someone add the holy shit tag because HOLY SHIT

**Comment by SaphireLattice** [comment]
> Okay it's very funny to say crayon but I really would prefer if you followed PR guidelines and explained the features your PR is implementing. Based on the videos it seems like it's some cool stuff!
> 
> Does the new UI play well with the rainbow crayon?

Fixed! And yeah, it works just fine, it's pretty much the same ol' UI, just touched up a lot.

File: Content.Shared/Crayon/SharedCrayonComponent.cs [hunk 1]
```diff
@@ -37,6 +37,17 @@ public CrayonColorMessage(Color color)
         }
     }
 
+    [Serializable, NetSerializable]
```
Comment (chromiumboy) [inline Content.Shared/Crayon/SharedCrayonComponent.cs:40]: This component has no documentation, would you mind adding some summary tags for the fields and the events?

File: Content.Client/Crayon/UI/CrayonWindow.xaml.cs [hunk 2]
```diff
@@ -44,51 +47,94 @@ private void SelectColor(Color color)
...
-                    Modulate = _color,
+                    Text = locName
                 };
-                button.OnPressed += ButtonOnPressed;
-                if (_selected == decal)
+
+                var grid = new GridContainer
                 {
-                    var panelContainer = new PanelContainer()
+                    Columns = 6,
```
[the remark this answers is not in the document]
Comment (SaphireLattice) [reply Content.Client/Crayon/UI/CrayonWindow.xaml.cs:90]: That would be lovely but I'm not sure how to do this right now. Probably should adjust the column Control eventually to allow for this sort of stuff automatically when given a `HorizontalExpand="True"`.

**Review by chromiumboy — APPROVED** [review]
Looks good to go

**Pull request merged**